lib/tree.rb

```diff
@@ -1,38 +1,60 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+RANDOM = Random.new;
+
+class Tree
+  attr_reader :height, :age, :apples, :alive
 
   def initialize
+    @age = 0
+    @height = 1
+    @apples = []
+    @alive = true
   end
 
   def age!
+    @age += 1
+    if @alive
```

Comment:
POODR has some great advice on why you might want to refer to ivars by their getters in most cases. No big deal just something to consider.